### PR TITLE
Feature/ingredient database

### PR DESCRIPTION
**Pull request by RagdollGoDown**
# Base Ingredient Database
## Description
- What has been changed?

We now have the base elements to integrate ingredient repositories. This pull comes with the implemented OpenFoodFacts repo for ingredients.

- Why was this change made?

To allow the development of features related to ingredients

## Checklist
- [x] Tests added.
- [x] Documentation updated.
- [x] All CI checks passed.
- [x] Linked issue/feature (if applicable): #68, #53

### CODE REVIEW

File: app/src/main/java/com/android/sample/model/ingredient/OpenFoodFactsIngredientRepository.kt [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+      onSuccess: (Ingredient?) -> Unit,
+      onFailure: (Exception) -> Unit
+  ) {
+    val url = "$openFoodFactsUrl/api/v2/product/$barCode"
+
+    val request =
+        Request.Builder()
+            .url(url)
+            // TODO: Add a proper User-Agent
+            .header("User-Agent", "PlateSwipe/1.0 (andre.cadet@epfl.ch)")
```
Comment (akshire) [inline app/src/main/java/com/android/sample/model/ingredient/OpenFoodFactsIngredientRepository.kt:34]: Here we need to set some general User-Agent, for example PlateSwipe

File: app/src/main/java/com/android/sample/model/ingredient/OpenFoodFactsIngredientRepository.kt [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+      onFailure: (Exception) -> Unit,
+      count: Int
+  ) {
+    val url = "$openFoodFactsUrl/cgi/search.pl?search_terms=$name&json=1&page_size=$count"
+
+    val request =
+        Request.Builder()
+            .url(url)
+            // TODO: Add a proper User-Agent
+            .header("User-Agent", "PlateSwipe/1.0 (andre.cadet@epfl.ch)")
```
Comment (akshire) [inline app/src/main/java/com/android/sample/model/ingredient/OpenFoodFactsIngredientRepository.kt:77]: Here again, you need to set a more general User-Agent that designs our app

File: app/src/main/java/com/android/sample/model/ingredient/OpenFoodFactsIngredientRepository.kt (hunk 1 above)
Comment (akshire) [inline app/src/main/java/com/android/sample/model/ingredient/OpenFoodFactsIngredientRepository.kt:34]: Here we need to set some general User-Agent, for example PlateSwipe

**Review by akshire — COMMENTED** [review]
Nice work ! It looks good overall :) You only need to change the two User Agent Profiles to make them more specific to our app.

**Review by 2Fick — APPROVED** [review]
### Overall Review
Great job on implementing the base elements for integrating ingredient repositories. The code is well-structured.

### Specific Comments
1. **User-Agent Header**:
   - In `OpenFoodFactsIngredientRepository.kt`:
     ```kotlin
     .header("User-Agent", "PlateSwipe/1.0 (plateswipe@gmail.com)")
     ```
     **Acknowledgement**: The User-Agent header has been successfully updated to a more appropriate and generic email.

### Summary
- The code changes are comprehensive and address the issues described in your PR.
- The User-Agent header has been updated as requested.

**Pull request merged**